tooling/internal/terraform/registry/archives.go

```diff
@@ -0,0 +1,62 @@
+package registry
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"io"
+
+	"github.com/gravitational/trace"
+)
+
+func repack(dst io.Writer, src io.Reader) error {
```

Comment:
Can you add a godoc explaining what repack does? I.e. what are the expected src and dst formats. Maybe include a link to the terraform docs explaining what the format for the provider is expected to be.